tox.ini

```diff
@@ -4,79 +4,103 @@
 # and then run "tox" from this directory.
 
 [tox]
+# To use a PEP 517 build-backend you are required to configure tox to use an isolated_build:
+# https://tox.readthedocs.io/en/latest/example/package.html
+isolated_build = True
+skipsdist = True
+
 envlist =
+    # always keep coverage-clean first
+    coverage-clean
     lint
     flake8
     mypy
-    manifest
+    docstr-coverage
     py
 
 [testenv]
+allowlist_externals =
+    poetry
 commands =
-    python -m pytest
-deps = 
-    linkml
-extras =
-    test
+    poetry run pytest
 description = Run unit tests with pytest. This is a special environment that does not get a name, and
               can be referenced with "py".
 
+[testenv:coverage-clean]
+deps = coverage
+skip_install = true
+commands = coverage erase
+
 [testenv:lint]
+deps =
+    black
+    isort
 skip_install = true
 commands =
-    black sssom/ tests/ setup.py
-    isort sssom/ tests/ setup.py
+    black src/ tests/
+    isort src/ tests/
+description = Run linters.
+
+[testenv:doclint]
 deps =
-    isort
-    black
-description = Run code formatters and linters.
+    rstfmt
+skip_install = true
+commands =
+    rstfmt docs/source/
+description = Run documentation linters.
 
 [testenv:flake8]
 skip_install = true
-commands =
-    flake8 sssom/ tests/ 
 deps =
+    darglint
     flake8<5.0.0
     flake8-black
-    flake8-colors
-    bandit
     flake8-bandit
-    pep8-naming
     flake8-bugbear
-    flake8-isort
+    flake8-colors
     flake8-docstrings
+    flake8-isort
+    flake8-print
     pydocstyle
-    darglint
-description = Run the flake8 code quality checker.
+commands =
+    flake8 src/ tests/
+description = Run the flake8 tool with several plugins (bandit, docstrings, import order, pep8 naming).
+
+#########################
+# Flake8 Configuration  #
+# (.flake8)             #
+#########################
+[flake8]
+max-line-length=120
+ignore =
+    DAR101 # Missing parameter(s) in Docstring: - with_git_hash
```

Comment:
Isn't this a bit central? How can we enforce this moving forward? Can we autogenerate docstrings for everything and leave it empty, so that in the future we have that check active?

Reply:
This error shows up in spite of parameters being present in the docstring. Not sure what it expects.

Reply:
removing these `ignores is a PR in itself. Could be assigned to someone who I wants to learn linting.